homeassistant/components/template/light.py

```diff
@@ -332,26 +324,39 @@ async def async_turn_on(self, **kwargs):
             self._temperature = kwargs[ATTR_COLOR_TEMP]
             optimistic_set = True
 
-        if ATTR_BRIGHTNESS in kwargs and self._level_script:
-            await self._level_script.async_run(
-                {"brightness": kwargs[ATTR_BRIGHTNESS]}, context=self._context
-            )
-        elif ATTR_COLOR_TEMP in kwargs and self._temperature_script:
-            await self._temperature_script.async_run(
-                {"color_temp": kwargs[ATTR_COLOR_TEMP]}, context=self._context
-            )
-        elif ATTR_WHITE_VALUE in kwargs and self._white_value_script:
-            await self._white_value_script.async_run(
-                {"white_value": kwargs[ATTR_WHITE_VALUE]}, context=self._context
-            )
-        elif ATTR_HS_COLOR in kwargs and self._color_script:
+        on_attrs = {}
+        if self._level_action:
+            on_attrs["brightness_new"] = False
+            on_attrs["brightness"] = self._brightness
+        if self._temperature_action:
+            on_attrs["color_temp_new"] = False
+            on_attrs["color_temp"] = self._color
+        if self._white_value_action:
+            on_attrs["white_value_new"] = False
+            on_attrs["white_value"] = self._white_value
+        if self._color_action:
+            on_attrs["hs_new"] = False
+            on_attrs["hs"] = self._color
+            on_attrs["h"] = int(self._color[0])
+            on_attrs["s"] = int(self._color[1])
+
+        if ATTR_BRIGHTNESS in kwargs and self._level_action:
+            on_attrs["brightness_new"] = True
+            on_attrs["brightness"] = kwargs[ATTR_BRIGHTNESS]
+        if ATTR_COLOR_TEMP in kwargs and self._temperature_action:
+            on_attrs["color_temp_new"] = True
+            on_attrs["color_temp"] = kwargs[ATTR_COLOR_TEMP]
+        elif ATTR_WHITE_VALUE in kwargs and self._white_value_action:
+            on_attrs["white_value_new"] = True
+            on_attrs["white_value"] = kwargs[ATTR_WHITE_VALUE]
+        elif ATTR_HS_COLOR in kwargs and self._color_action:
```

Comment:
What happen if we set white value and hs color at the same time?

Reply:
I wasn't sure how to handle this since multiple color options would be conflicting. I guess it should either error, do nothing, or pick the first one. The original code (before this pull) also did not handle this.

Reply:
When I saw your issue I thought immediately to this part of code. I don't see any reason to keep elif instead of if block. I have a light that's support temp, color and white.

Reply:
OK, I will remove the if block.

Though I am confused what the expected output is if you set all of these simultaneously:
```yaml
service: light.turn_on
entity_id: light.my_light
data:
  hs_color: [124, 213],
  white_value: 200,
  color_temp: 250
```
What should a light do in this situation? These three values represent 3 different colors. I suppose by removing the if block the user will be allowed to select what happens with conflicts.